lib/improver/tests/test_weights_ChooseDefaultWeightsLinear.py

```diff
@@ -0,0 +1,191 @@
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the weights.ChooseDefaultWeightsLinear plugin."""
+
+
+import unittest
+
+from cf_units import Unit
+from iris.coords import AuxCoord, DimCoord
+from iris.cube import Cube
+from iris.tests import IrisTest
+import iris
+import numpy as np
+
+from improver.weights import ChooseDefaultWeightsLinear as LinearWeights
+
+
+def add_realizations(cube, num):
+    """ Create num realizations of input cube
+        Args:
+            cube =iris.cube.Cube - input cube
+            num = integer - Number of realizations
+        Returns
+            cubeout = iris.cube.Cube - copy of cube with num realizations added
+    """
+    cubelist = iris.cube.CubeList()
+    for i in range(0, num):
+        newcube = cube.copy()
+        new_ensemble_coord = iris.coords.AuxCoord(i,
+                                                  standard_name='realization')
+        newcube.add_aux_coord(new_ensemble_coord)
+        cubelist.append(newcube)
+    cubeout = cubelist.merge_cube()
+    return cubeout
+
+
+class TestChooseDefaultWeightsLinear(IrisTest):
+    """ Test the Default Linear Weights plugin """
+
+    def setUp(self):
+        data = np.zeros((2, 2, 2))
+        data[0][:][:] = 0.0
+        data[1][:][:] = 1.0
+        cube = Cube(data, standard_name="precipitation_amount",
+                    units="kg m^-2 s^-1")
+        cube.add_dim_coord(DimCoord(np.linspace(-45.0, 45.0, 2), 'latitude',
+                                    units='degrees'), 1)
+        cube.add_dim_coord(DimCoord(np.linspace(120, 180, 2), 'longitude',
+                                    units='degrees'), 2)
+        time_origin = "hours since 1970-01-01 00:00:00"
+        calendar = "gregorian"
+        tunit = Unit(time_origin, calendar)
+        cube.add_aux_coord(AuxCoord([402192.5, 402193.5],
+                                    "time", units=tunit), 0)
+        dummy_scalar_coord = iris.coords.AuxCoord(1,
+                                                  long_name='scalar_coord',
+                                                  units='no_unit')
+        cube.add_aux_coord(dummy_scalar_coord)
+        self.cube = cube
+
+    def test_basic(self):
+        """ Test that the plugin retuns an array of weights """
+        coord = "time"
+        plugin = LinearWeights()
+        result = plugin.process(self.cube, coord)
+        self.assertIsInstance(result, np.ndarray)
+
+    def test_array_sum_equals_one(self):
+        """ Test that the resulting weights add up to one """
```

Comment:
The question is whether this is a title (no full stop) or a Sentence (full stop) or a phrase (no full stop) and whether it really matters. I don't think this is a useful use of my time.